sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java

```diff
@@ -0,0 +1,290 @@
+                            new Identity().withType(IdentityType.NONE).withUserAssignedIdentities(null));
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Initialize storage account's identity property.
+     *
+     * @param identityType the identity type to set
+     */
+    private void initStorageAccountIdentity(IdentityType identityType, Boolean isWithout) {
+        if (!identityType.equals(IdentityType.USER_ASSIGNED)
+            && !identityType.equals(IdentityType.SYSTEM_ASSIGNED)) {
+            throw logger.logExceptionAsError(new IllegalArgumentException("Invalid argument: " + identityType));
+        }
+        if (!isWithout) {
+            if (this.storageAccount.isInCreateMode()) {
+                if (Objects.isNull(this.storageAccount.createParameters.identity())) {
+                    this.storageAccount.createParameters.withIdentity(new Identity());
+                }
+                if (Objects.isNull(this.storageAccount.createParameters.identity().type())
+                    || this.storageAccount.createParameters.identity().type().equals(IdentityType.NONE)
+                    || this.storageAccount.createParameters.identity().type().equals(identityType)) {
+                    this.storageAccount.createParameters.identity().withType(identityType);
+                } else {
+                    this.storageAccount.createParameters.identity().withType(IdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED);
+                }
+            } else {
+                if (Objects.isNull(this.storageAccount.updateParameters.identity())) {
+                    this.storageAccount.updateParameters.withIdentity(this.storageAccount.innerModel().identity());
+                }
+                if (Objects.isNull(this.storageAccount.updateParameters.identity())) {
+                    this.storageAccount.updateParameters.withIdentity(new Identity());
+                }
+                if (Objects.isNull(this.storageAccount.updateParameters.identity().type())
+                    || this.storageAccount.updateParameters.identity().type().equals(IdentityType.NONE)
+                    || this.storageAccount.updateParameters.identity().type().equals(identityType)) {
+                    this.storageAccount.updateParameters.identity().withType(identityType);
+                } else {
+                    this.storageAccount.updateParameters.identity().withType(IdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED);
+                }
+            }
+        } else {
+            if (Objects.isNull(this.storageAccount.updateParameters.identity())) {
+                this.storageAccount.updateParameters.withIdentity(this.storageAccount.innerModel().identity());
+            }
+            if (IdentityType.SYSTEM_ASSIGNED.equals(identityType)) {
+                if (IdentityType.SYSTEM_ASSIGNED.equals(this.storageAccount.innerModel().identity().type())) {
```

Comment:
This line seems still need to check `storageAccount.updateParameters.identity().type()`?

Reply:
`if (IdentityType.SYSTEM_ASSIGNED.equals(this.storageAccount.innerModel().identity().type()))` corrected to `if (IdentityType.SYSTEM_ASSIGNED.equals(this.storageAccount.updateParameters.identity().type()))`